src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizConfiguration.java

```diff
@@ -0,0 +1,109 @@
+     */
+    default void reconnectJSONIgnoreAttributes() {// iterate through quizQuestions to add missing pointer back to quizExercise
+        // Note: This is necessary because of the @IgnoreJSON in question and answerOption
+        // that prevents infinite recursive JSON serialization.
+        for (QuizQuestion quizQuestion : getQuizQuestions()) {
+            if (quizQuestion.getId() != null) {
+                setQuestionParent(quizQuestion);
+                // reconnect QuestionStatistics
+                if (quizQuestion.getQuizQuestionStatistic() != null) {
+                    quizQuestion.getQuizQuestionStatistic().setQuizQuestion(quizQuestion);
+                }
+                // do the same for answerOptions (if quizQuestion is multiple choice)
+                if (quizQuestion instanceof MultipleChoiceQuestion mcQuestion) {
+                    MultipleChoiceQuestionStatistic mcStatistic = (MultipleChoiceQuestionStatistic) mcQuestion.getQuizQuestionStatistic();
+                    // reconnect answerCounters
+                    for (AnswerCounter answerCounter : mcStatistic.getAnswerCounters()) {
+                        if (answerCounter.getId() != null) {
+                            answerCounter.setMultipleChoiceQuestionStatistic(mcStatistic);
+                        }
+                    }
+                    // reconnect answerOptions
+                    for (AnswerOption answerOption : mcQuestion.getAnswerOptions()) {
+                        if (answerOption.getId() != null) {
+                            answerOption.setQuestion(mcQuestion);
+                        }
+                    }
+                }
+                if (quizQuestion instanceof DragAndDropQuestion dragAndDropQuestion) {
+                    DragAndDropQuestionStatistic dragAndDropStatistic = (DragAndDropQuestionStatistic) dragAndDropQuestion.getQuizQuestionStatistic();
+                    // reconnect dropLocations
+                    for (DropLocation dropLocation : dragAndDropQuestion.getDropLocations()) {
+                        if (dropLocation.getId() != null) {
+                            dropLocation.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect dragItems
+                    for (DragItem dragItem : dragAndDropQuestion.getDragItems()) {
+                        if (dragItem.getId() != null) {
+                            dragItem.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect correctMappings
+                    for (DragAndDropMapping mapping : dragAndDropQuestion.getCorrectMappings()) {
+                        if (mapping.getId() != null) {
+                            mapping.setQuestion(dragAndDropQuestion);
+                        }
+                    }
+                    // reconnect dropLocationCounters
```

Comment:
This is three times the same for loop. You could generalize this for loop using a function with a generic function parameter.

Reply:
loop is moved to a generalized method.